app/actions/views/channel.js

```diff
@@ -645,15 +713,19 @@ export function loadChannelsForTeam(teamId, skipDispatch = false) {
                             console.log('Could not retrieve channel members roles for the user');
                         }
                     }
-
-                    dispatch(batchActions(actions, 'BATCH_LOAD_CHANNELS_FOR_TEAM'));
                 }
 
                 // Fetch needed profiles from channel creators and direct channels
                 dispatch(loadSidebar(data));
 
                 dispatch(loadUnreadChannelPosts(data.channels, data.channelMembers));
             }
+
+            dispatch(loadGroupData());
+        }
+
+        if (actions.length) {
+            dispatch(batchActions(actions, 'BATCH_LOAD_CHANNELS_FOR_TEAM'));
```

Comment:
we are we moving this from were it was? there is a flag to determine if we need to dispatch or not.

Reply:
Okay! My bad 😄 Done.